packages/caravan-fees/src/rbf.ts

```diff
@@ -0,0 +1,448 @@
+import { PsbtV2 } from "@caravan/psbt";
+import { Network } from "@caravan/bitcoin";
+import BigNumber from "bignumber.js";
+import { createOutputScript } from "./utils";
+import { DEFAULT_DUST_THRESHOLD, RBF_SEQUENCE } from "./constants";
+import { RbfOptions, FeeRate, UrgencyLevel } from "./types";
+
+/**
+ * RbfTransaction Class
+ *
+ * This class implements the Replace-By-Fee (RBF) functionality for Bitcoin transactions,
+ * adhering to BIP125 (https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki).
+ * It allows for the creation of replacement transactions with higher fees, either to
+ * accelerate confirmation or to cancel a previous transaction.
+ *
+ * The class supports both transaction acceleration and cancellation, with configurable
+ * fee rates based on network conditions and user-defined urgency levels.
+ *
+ * Key Features:
+ * - Supports PSBTv2 and PSBTv0 formats
+ * - Implements BIP125 rules for RBF
+ * - Allows fee subtraction from specific outputs or proportional distribution
+ * - Supports additional input selection for fee coverage
+ * - Provides transaction cancellation functionality
+ *
+ * Usage:
+ * const rbfTx = new RbfTransaction(psbt, feeMarket, options, network);
+ * const acceleratedPsbt = rbfTx.prepareAccelerated();
+ * // or
+ * const canceledPsbt = rbfTx.prepareCanceled(destinationAddress);
+ */
+export class RbfTransaction {
+  private originalPsbt: PsbtV2;
+  private newPsbt: PsbtV2;
+  private options: RbfOptions;
+  private feeMarket: FeeRate;
+  private urgencyMultipliers: Record<UrgencyLevel, number>;
+  private network: Network;
+
+  /**
+   * Constructor for RbfTransaction
+   *
+   * @param {PsbtV2 | string | Buffer} psbt - The original PSBT to be replaced
+   * @param {FeeRate} feeMarket - Current fee market rates
+   * @param {RbfOptions} options - Configuration options for the RBF transaction
+   * @param {Network} network - The Bitcoin network (mainnet, testnet, etc.)
+   *
+   * @throws {Error} If the PSBT is not in a valid format or not signaling RBF
+   */
+  constructor(
+    psbt: PsbtV2 | string | Buffer,
+    feeMarket: FeeRate,
+    options: RbfOptions,
+    network: Network,
```

Comment:
I think it's cleaner to have these all as a single options object. This can be a confusing api to work with in my opinion. You have to know the order these get passed in and some are nested in an object and others are at the top level. Another advantage of putting it all in a single object (all typed) is clean, self documenting instantiation (whether or not with object property shorthand depending on context):

``` typescript
const tx = new RbfTransaction({
  network, 
  targetFeeRate: networkFees.fast,
  psbt,
  ...rest
})
```

instead of 

``` typescript
const tx = new RbfTransaction(psbt, networkFees.fast, { ...rest }, MAINNET)
```

Reply:
Addressed in commit [Refactored RBF and CPFP classes, introduced TransactionAnalyzer class](https://github.com/caravan-bitcoin/caravan/commit/77f0cbeb67a274b0aea5a69e87168b9ac1e2304a)